### ISSUE TITLE
Make authorize! return the object that was passed into the method.

### ISSUE THREAD

**Issue by jnicklas** [elabs/pundit#375]
Right now it always returns `true`, this is somewhat useless. By having it return the object, we can do all kinds of useful patterns, like `render json: authorize(@foo)`.

**Comment by ce07c3** [comment]
I would suggest that the policies themselves can return more than just true. One use case is evaluating rules in the policies - sometimes you need to know the results.

**Comment by formigarafa** [comment]
@ce07c3 , I don't think that would be a good idea. We could end up with another service library doing lots of stuff instead of what it is supposed to do: check policies. That's exactly the reason of policy methods end in question marks. But if you think you need something more, take a look at active_interaction , that might give you what you are asking for here. :)

**Comment by ce07c3** [comment]
@formigarafa There lies a lot of value in returning more than just true or false once your application needs grow. For instance, what if I wish to whitelist parameters for responses to API requests? If my policy would return a garafa result including the set of permitted attributes, I could pass this on to subsequent serializers. :)

**Comment by formigarafa** [comment]
@ce07c3,
but that is exactly what a meant by doing too much stuff. It would become unclear what pundit is supposed to do. And while it might seem a good idea on a initial moment, when your application grow it will be even more difficult reuse a police/permitted-params thing. And like I said, you can still use a third class to achieve what you suggested. 
A PolicyAwareParameters class could get a policy instance and return your whitelisted params. But I think it makes more sense a class having pundit as a dependency to do something more than change it to have more responsibilities than any class should. This would probably lead to a future extraction of policy permission class from the policy class to ensure each one have a single responsibility.

BTW, I like your idea of having something to return the permitted params according to a policy.
It would be very helpful.

**Comment by ce07c3** [comment]
I saw code or issues about caching policies. Since policies can only return true or false, I'd have to set data on the policy itself if I were to create a `PolicyAwareParameters` class, rendering them thread unsafe.

Also, I'd like to adhere to DRY principles so specifying the attributes in the policy feels right.

*active_interaction* looks nice, I came up with a few ideas on where to use it in my app (not for authorization, though =D)
